README.md

```diff
@@ -56,51 +64,86 @@ Improtant files:
 
-8. qemu-img convert -f raw -O qcow2 rootfs.ext2 rootfs.qcow2
+9. 
+```
+cd ./output/images
+qemu-img convert -f raw -O qcow2 rootfs.ext2 rootfs.qcow2`
+```
+
+> Note: rootfs.ext4 is just a symlink to rootfs.ext2
 
 ## Launch the vm
-9. sudo qemu-system-x86_64 -enable-kvm -cpu host -s -kernel bzImage  -m 2048 -hda rootfs.qcow2 -append "root=/dev/sda rw nokaslr" -net nic,model=virtio -net user,hostfwd=tcp::5555-:22
 
-Lets explain our command:
-* -enable-kvm -> kvm is a virtualization solution for linux which use hardware virtualization extensions, we will use it in order to improve the vm performence
+Buildroot has created a script at `./output/images/start-qemu.sh`. We'll
+add our own parameters to accelerate performance and setups connections.
+
+
+10. Add the next flags
+
+* -monitor unix:qemu-monitor-socket,server,nowait -> creates a socket file named `qemu-monitor-socket` to which we'll connect with socat for the qemu monitoring
+* -enable-kvm -> kvm is a virtualization solution for linux which use hardware virtualization extensions, we will use it in order to improve the vm performance
 * -cpu host -> use host cpu, we will use it in order to improve the vm performence
 * -s -> qemu will open a gdbserver on TCP port 1234
-* -kernel -> path to the compiled kernel image
-* -m -> amount of memory of the vm
+* -m 2048 -> amount of memory of the vm (2mb in our example)
+
+replace `-drive file=rootfs.ext2,ifvirtio,format=raw` with `-hda rootfs.qcow2`
 * -hda -> path to the root filesystem image in our case the rootfs 
+
 * -append -> send command line arguments to the linux kernel
+replace `root=/dev/vda` with `root=/dev/sda`
+add `rw nokaslr` 
+
 * -net nic,model=virtio -> connect a network interface
-* -net user,hostfwd=tcp::5555-:22 -> in order to use ssh we will forward tcp traffic from host port 5555 to guest port 22
+replace `-net user` with `-net user,hostfwd=tcp::5555-:22`
+this will forward tcp traffic from host port 5555 to guest port 22
+which allows us to use ssh
+
+your final command in `start_qemu.sh` should look something like this:
+
+`exec qemu-system-x86_64 -s -enable-kvcm -cpu host -m 2048 -M pc -kernel bzImage -hda rootfs.qcow2 -append "rootwait root=/dev/vda console=tty1 console=ttyS0
+-net nic,model=virtio -net user,hostfwd=tcp::5555-:22`
+
+Now we can launch our vm:
+
+11. ./start-qemu.sh
+
+#### SSH to guest
+`ssh root@localhost -p 5555`
 
-## Setup ssh (nice to have)
-After launcing the vm you should see buildroot login the username is root and use the password which we configured earlier.
-**Add "PermitRootLogin yes" to /etc/ssh/sshd_config in order to enable root login using ssh.**
-Now you can open the terminal on the host and execute:
-ssh -p 5555 root@localhost
 
-## Take a snapshot
-We will switch to qemu console in order to take the snapshot. 
+## Snapshots
+In order to take snapshots we'll connect to the qemu monitor 
+11. socat stdio,echo=0,icanon=0 unix-connect:qemu-monitor-socket
```

Comment:
Can you please explain why it is needed

Reply:
loading snapshots requires usage of the qemu monitor. This allows connecting to the monitor via terminal (allows ssh). Without this you have to use gui and shortkeys which is not possible through an ssh connection

Reply:
Does it a command to execute?

Reply:
what?